### PR TITLE
Camera API tweaks

### PR DESCRIPTION
**Pull request by matteblair**
- Change "ease" member in Map from a vector to a single unique_ptr, since we only have one at a time.
- Allow specifying separate padding values for each side of the view when fitting a camera to a region.
- Fix some JNI warnings in Android Studio

### CODE REVIEW

File: core/src/map.cpp [hunk 1]
```diff
@@ -760,27 +760,36 @@ float Map::getTilt() {
...
+    const MapProjection& projection = view.getMapProjection();
+
+    // Convert the bounding coordinates into Mercator meters.
     glm::dvec2 aMeters = projection.LonLatToMeters(glm::dvec2(_a.longitude, _a.latitude));
     glm::dvec2 bMeters = projection.LonLatToMeters(glm::dvec2(_b.longitude, _b.latitude));
+    glm::dvec2 dMeters = glm::abs(aMeters - bMeters);
 
-    double tileSize = projection.TileSize() * impl->view.pixelScale();
-    double buffer = _buffer * 2 * impl->view.pixelScale();
+    // Calculate the inner size of the view that the bounds must fit within.
```
Comment (tallytalwar) [inline core/src/map.cpp]: 👍 for comments.

File: core/include/tangram/map.h [hunk 2]
```diff
@@ -230,8 +230,9 @@ class Map {
     // Get the tilt angle of the view in radians; 0 corresponds to straight down
     float getTilt();
 
-    // Get the CameraPosition that encloses the bounds given by _a and _b
-    CameraPosition getEnclosingCameraPosition(LngLat a, LngLat b, int buffer);
+    // Get the CameraPosition that encloses the bounds given by _a and _b and
+    // leaves at least the given amount of padding on each side (in logical pixels).
+    CameraPosition getEnclosingCameraPosition(LngLat _a, LngLat _b, int _leftPad, int _topPad, int _rightPad, int _bottomPad);
```
Comment (tallytalwar) [inline core/include/tangram/map.h]: Will it be better (cleaner) to pass in std::array here instead of explicit 4 padding dimensions?

Comment (matteblair) [reply core/include/tangram/map.h]: I'm pretty sure I'd mess up the order of these arguments if they weren't named. The better solution for consistency is probably to define a padding struct with named members.

Comment (tallytalwar) [reply core/include/tangram/map.h]: Yeah, maybe its a small fish in big ocean, but we do lose some cache coherence by breaking the well formed memory struct for these paddings.

File: core/include/tangram/map.h [hunk 3]
```diff
@@ -100,6 +100,18 @@ struct CameraPosition {
     float tilt = 0;
 };
 
+struct EdgePadding {
+    int left = 0;
```
Comment (tallytalwar) [inline core/include/tangram/map.h]: Can we have negative padding? Or rather should we allow negative padding?

Comment (matteblair) [reply core/include/tangram/map.h]: Yeah it's possible. It's not very useful but some application might need it - might as well allow it.

**Pull request merged**